chain/client/src/client.rs

```diff
@@ -543,6 +541,19 @@ impl ClientActor {
             self.process_blocks_with_missing_chunks(ctx, block_hash);
         }
 
+        // Update latest known height if given block is indeed latest known.
+        let this_latest_known =
+            LatestKnown { height: block.header.inner.height, seen: to_timestamp(Utc::now()) };
+        let latest_known = self
+            .chain
+            .mut_store()
+            .get_latest_known()
+            .ok()
+            .unwrap_or_else(|| this_latest_known.clone());
```

Comment:
This seems to be a hacky way to avoid an extra condition. Naturally, if we `unwrap_or_else` here, the next condition doesn't trigger, and `latest_known` is never used.
If this `unwrap_or_else` is removed, the computation of `this_latest_known` can be moved inside the condition below.